### PR TITLE
uploadclient: upload method Mario - #3084

### PR DESCRIPTION
**Pull request by TomasJavurek**
uploadclient: upload method Mario - #3084

### CODE REVIEW

File: lib/rucio/client/uploadclient.py [hunk 1]
```diff
@@ -100,6 +101,7 @@ def upload(self, items, summary_file_path=None, traces_copy_out=None):
         :raises NotAllFilesUploaded: if not all files were successfully uploaded
         """
         logger = self.logger
+        logger.info(str(items))
```
Comment (mlassnig) [inline lib/rucio/client/uploadclient.py]: should be .debug()

File: lib/rucio/client/uploadclient.py [hunk 2]
```diff
@@ -234,19 +238,17 @@ def upload(self, items, summary_file_path=None, traces_copy_out=None):
...
-                    file['upload_result'] = state
+                    self._upload_item(rse_settings=rse_settings,
+                                      lfn=lfn,
+                                      source_dir=file['dirname'],
+                                      force_scheme=cur_scheme,
+                                      force_pfn=pfn,
+                                      transfer_timeout=file.get('transfer_timeout'),
+                                      delete_existing=delete_existing,
+                                      sign_service=sign_service)
+                    logger.debug('Upload done.')
```
Comment (mlassnig) [inline lib/rucio/client/uploadclient.py:250]: .info() ?

File: lib/rucio/client/uploadclient.py [hunk 3]
```diff
@@ -495,6 +497,150 @@ def _convert_file_for_api(self, file):
...
+
+        # Getting pfn
+        pfn = None
+        readpfn = None
+        try:
+            pfn = list(protocol_write.lfns2pfns(make_valid_did(lfn)).values())[0]
+            readpfn = list(protocol_read.lfns2pfns(make_valid_did(lfn)).values())[0]
+        except Exception as error:
+            logger.warning('Failed to create PFN for LFN: %s' % lfn)
+            logger.warning(str(error))
```
Comment (mlassnig) [inline lib/rucio/client/uploadclient.py]: .debug()

File: lib/rucio/client/uploadclient.py [hunk 4]
```diff
@@ -495,6 +497,150 @@ def _convert_file_for_api(self, file):
...
+
+        # Auth. mostly for object stores
+        if sign_service:
+            pfn = self.client.get_signed_url(sign_service, 'write', pfn)       # NOQA pylint: disable=undefined-variable
+            readpfn = self.client.get_signed_url(sign_service, 'read', pfn)    # NOQA pylint: disable=undefined-variable
+
+        # Create a name of tmp file if renaming operation is supported
+        pfn_tmp = '%s.rucio.upload' % pfn if protocol_write.renaming else pfn
+        readpfn_tmp = '%s.rucio.upload' % readpfn if protocol_write.renaming else readpfn
+
```
Comment (mlassnig) [inline lib/rucio/client/uploadclient.py:561]: these two could go into a  .debug() ?

File: lib/rucio/client/uploadclient.py [hunk 5]
```diff
@@ -495,6 +497,150 @@ def _convert_file_for_api(self, file):
...
+                    if stats['adler32'] != lfn['adler32']:
+                        raise RucioException('Checksum mismatch. Source: %s Destination: %s' % (lfn['adler32'], stats['adler32']))
+
+        except Exception as error:
+            raise error
+
+        # The upload finished successful and the file can be renamed
+        try:
+            if protocol_write.renaming:
+                protocol_write.rename(pfn_tmp, pfn)
```
Comment (mlassnig) [inline lib/rucio/client/uploadclient.py:617]: maybe add .debug('Renaming file from %s to %s') ?

**Review by mlassnig — APPROVED** [review]

File: lib/rucio/client/uploadclient.py [hunk 6]
```diff
@@ -234,22 +239,20 @@ def upload(self, items, summary_file_path=None, traces_copy_out=None):
...
-                    success = state['success']
-                    file['upload_result'] = state
+                    self._upload_item(rse_settings=rse_settings,
+                                      lfn=lfn,
+                                      source_dir=file['dirname'],
+                                      force_scheme=cur_scheme,
+                                      force_pfn=pfn,
+                                      transfer_timeout=file.get('transfer_timeout'),
+                                      delete_existing=delete_existing,
+                                      sign_service=sign_service)
```
Comment (bari12) [inline lib/rucio/client/uploadclient.py:249]: I think you are missing the auth_token (OIDC) here. That functionality was introduced to the rse manager by @gumond rather recently, but I think this OR would essentially take the passthrough out of the upload workflow again. Or am I missing something?

Comment (TomasJavurek) [reply lib/rucio/client/uploadclient.py:249]: the auth. is resoved when creating a protocol instance, not at the upload itself. But I'll rather cehck that.

**Pull request merged**